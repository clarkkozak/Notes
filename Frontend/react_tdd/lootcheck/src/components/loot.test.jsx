import React from 'react'
import { mount } from 'enzyme'
import { Loot } from './loot'

describe('Loot', () => {
  const mockFetchBitcoin = jest.fn()
  let props = {
    balance: 10,
    bitcoin: { bpi: { USD: { rate: '1,000' } } },
    fetchBitcoin: mockFetchBitcoin
  }

  const loot = mount(<Loot {...props} />)

  it('Renders Properly', () => {
    expect(loot).toMatchSnapshot()
  })

  describe('when mounted', () => {

    it('dispatches the `fetchBitcoin() method it receives from props', () => {
      expect(mockFetchBitcoin).toHaveBeenCalled()
    })
  })

  // describe('when there are valid bitcoin props', () => {
  //   it('displays the correct bitcoin value', () => {
  //     expect(loot.find('h3').text()).toEqual('Bitcoin Balance: 0.01')
  //   })
  // })
  
  
})