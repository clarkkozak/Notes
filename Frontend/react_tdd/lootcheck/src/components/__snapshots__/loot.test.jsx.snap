// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot Renders Properly 1`] = `ReactWrapper {}`;
